packages/tools/src/stateManagement/annotation/FrameOfReferenceSpecificAnnotationManager.ts

```diff
@@ -270,6 +275,13 @@ class FrameOfReferenceSpecificAnnotationManager implements IAnnotationManager {
         delete annotations[groupKey];
       }
     }
+
+    if (annotationToRemove !== null) {
+      triggerEvent(eventTarget, csToolEvents.ANNOTATION_REMOVED, {
+        annotation: annotationToRemove,
+        annotationManagerUID: this.uid,
+      });
+    }
```

Comment:
If you examine other sections in this file, we don't publish events here but rather in the `annotationState`. Could you perhaps follow the same pattern?